TreesForTravel/screens/HomeScreen.js

```diff
@@ -0,0 +1,105 @@
+import React from 'react';
+import { StyleSheet, Text, View, TextInput, Image, Dimensions } from 'react-native';
+import { connect } from 'react-redux';
+import { getFlights, postFlight } from '../Redux/actions.js';
+import CitySearch from '../components/citySearch.js';
+import Forest from '../components/forest.js';
+import SummaryBar from '../components/summaryBar.js';
+import Cities from '../data/cities.json';
+import DataFunctions from '../dataProcessing/functions.js'
+
+class HomeScreen extends React.Component {
+
+  constructor(props) {
+     super(props);
+     this.state = {
+       from: null,
+       to: null,
+       visible: false
+     }
+   }
+
+  static navigationOptions = {
+    title: 'My Stats',
+  };
+
+  addFlight = () => {
+    this.setState({
+      from: null,
+      to: null
+    });
+    const flightData = DataFunctions.analyseFlight(this.props.selectedOrigin, this.props.selectedDestination);
+    fetch('http://192.168.1.187:3000/flights', {
+      method: 'POST',
+      headers: {Accept: 'application/json', 'Content-Type': 'application/json'},
+      body: JSON.stringify(flightData),
+      })
+      .then(response => response.json())
```

Comment:
Same thing, buy having a general api client you can centralize all formatting, and to avoid repeating `response.json()`.